### PR TITLE
Port Basis tests to use doctest.

### PR DESCRIPTION
**Pull request by fire**
Test basis.

### CODE REVIEW

File: tests/test_basis.h [hunk 1]
```diff
@@ -31,10 +31,258 @@
...
+			ret.set_euler_zxy(p_rotation);
+			break;
+
+		case EulerZYX:
+			ret.set_euler_zyx(p_rotation);
+			break;
+
+		default:
+			// If you land here, Please integrate all rotation orders.
+			REQUIRE_MESSAGE(false, "This is not unreachable.");
```
Comment (Xrayez) [inline tests/test_basis.h:88]: You can use `FAIL` with a message which fails an entire test suite without running any remaining assertions.
And probably would better to encode the comment directly in there:
```suggestion
			FAIL("If you land here, please integrate all rotation orders.");
```

File: tests/test_basis.h [hunk 2]
```diff
@@ -31,10 +31,258 @@
...
+
+		case EulerZXY:
+			return p_rotation.get_euler_zxy();
+
+		case EulerZYX:
+			return p_rotation.get_euler_zyx();
+
+		default:
+			// If you land here, Please integrate all rotation orders.
+			REQUIRE_MESSAGE(false, "This is not unreachable.");
```
Comment (Xrayez) [inline tests/test_basis.h:116]: Same as above.
```suggestion
			FAIL("If you land here, please integrate all rotation orders.");
```

**Comment by Xrayez** [comment]
I think `String` and other Godot datatype conversion could be handled by implementing our custom console reporter #40890 (I've only provided a stub implementation of it) or other [stringification methods](https://github.com/onqtam/doctest/blob/master/doc/markdown/stringification.md).

File: tests/test_basis.h [hunk 3]
```diff
@@ -31,10 +31,258 @@
...
 namespace TestBasis {
-MainLoop *test();
+
+enum RotOrder {
+	EulerXYZ,
+	EulerXZY,
+	EulerYZX,
+	EulerYXZ,
+	EulerZXY,
+	EulerZYX
```
Comment (aaronfranke) [inline tests/test_basis.h:48]: Only YXZ is actually used, we can remove all the others. To be more precise, they're not exposed, and their only usage in the engine is in the test cases.

Comment (fire) [reply tests/test_basis.h:48]: Perhaps we expose them! I remember this is a prerequisite for maya fbx support.

Comment (Xrayez) [reply tests/test_basis.h:48]: If something is mathematically possible to calculate, I believe the tests should cover those as well, to provide more robustness, we may get even more stable `YXZ` computation as a byproduct.

Comment (fire) [reply tests/test_basis.h:48]: My decision is keep these tests because they're mathematically possible.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**